src/braket/circuits/noise_model/noise_model.py

```diff
@@ -365,6 +401,51 @@ def from_dict(cls, noise_dict: dict) -> NoiseModel:
         )
         return model
 
+    def _apply_measure_noise(self, circuit: Circuit) -> Circuit:
+        """Apply measure noise to the circuit.
+
+        Args:
+            circuit (Circuit): The circuit to apply measure noise to.
+
+        Returns:
+            Circuit: The circuit with measure noise applied.
+        """
+        result = Circuit()
+        i = 0
+        n = len(circuit.instructions)
+        while i < n:
+            instr = circuit.instructions[i]
+            if isinstance(instr.operator, Measure):
+                # Collect contiguous block of measure instructions
+                measure_block = []
+                while i < n and isinstance(circuit.instructions[i].operator, Measure):
+                    measure_block.append(circuit.instructions[i])
+                    i += 1
+                # Collect all qubits being measured in this block
+                all_measured_qubits = []
+                for m in measure_block:
+                    all_measured_qubits.extend(list(m.target))
+                # Insert all noise for all measured qubits before all measure instructions
+                for noise_instruction in self._instructions:
+                    if isinstance(noise_instruction.criteria, MeasureCriteria):
+                        if noise_instruction.criteria._qubits is None:
+                            for qubit in all_measured_qubits:
+                                result.add_instruction(Instruction(noise_instruction.noise, qubit))
+                        else:
+                            for qubit in all_measured_qubits:
+                                if qubit in noise_instruction.criteria._qubits:
+                                    result.add_instruction(Instruction(noise_instruction.noise, qubit))
+                # Add all measure instructions
+                for m in measure_block:
+                    result.add_instruction(m)
+            else:
+                result.add_instruction(instr)
+                i += 1
+        # Add all result types
```

Comment:
thanks for considering the result types! But in fact, the result types and `measure` are not compatible. we can remove this part.